pkgs/development/libraries/xsimd/default.nix

```diff
@@ -11,7 +11,9 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ cmake ];
 
-  cmakeFlags = [ "-DBUILD_TESTS=ON" ];
+  cmakeFlags = [
+    "-DBUILD_TESTS=${if (doCheck && stdenv.hostPlatform == stdenv.buildPlatform) then "ON" else "OFF"}"
```

Comment:
Is this a common pattern? Do we want to implicitly disable `doCheck` at the derivation level, rather than require the caller to disable `doCheck` and fail if they didn't?

Reply:
> Is this a common pattern? Do we want to implicitly disable `doCheck` at the derivation level, rather than require the caller to disable `doCheck` and fail if they didn't?

`cmake`'s setup-hook already performs a similar logic:

https://github.com/nixos/nixpkgs/blob/bec27fabee7ff51a4788840479b1730ed1b64427/pkgs/development/tools/build-managers/cmake/setup-hook.sh#L106-L108

It's just that specifically this package chose a different naming for the cmake flag that controls the tests build - `BUILD_TESTS` vs `BUILD_TESTING`.